### PR TITLE
Convert gBattleStringsTable to COMPOUND_STRINGs

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Supersedes #5598

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

**Comment by AlexOn1ine** [comment]
Just in case nobody else starts to suddenly review it, I planning to review this tomorrow.

**Comment by ShinyDragonHunter** [comment]
Might not be needed for this PR but there's a function in `battle_setup.c` named `ReturnEmptyStringIfNull` which returns an empty string if the string pointer passed to it is `NULL`.

It could be useful for places you use `COMPOUND_STRING` as it means you might not always need to put a pointer to an empty string in string tables (or if you changed trainer names to use `COMPOUND_STRING` for example) and could also come in handy for places that use text but said text hasn't been made yet

File: src/battle_message.c [hunk 1]
```diff
@@ -470,1121 +143,747 @@ const u8 *const gStatNamesTable[NUM_BATTLE_STATS] =
...
+    [STRINGID_ITDOESNTAFFECT]                       = COMPOUND_STRING("It doesn't affect {B_DEF_NAME_WITH_PREFIX2}…"),
+    [STRINGID_ATTACKERFAINTED]                      = COMPOUND_STRING("{B_ATK_NAME_WITH_PREFIX} fainted!\p"),
+    [STRINGID_TARGETFAINTED]                        = COMPOUND_STRING("{B_DEF_NAME_WITH_PREFIX} fainted!\p"),
+    [STRINGID_PLAYERGOTMONEY]                       = COMPOUND_STRING("You got ¥{B_BUFF1} for winning!\p"),
+    [STRINGID_PLAYERWHITEOUT]                       = COMPOUND_STRING("You have no more Pokémon that can fight!\p"),
+#if B_WHITEOUT_MONEY >= GEN_4
+    [STRINGID_PLAYERWHITEOUT2]                      = COMPOUND_STRING("You panicked and dropped ¥{B_BUFF1}…\pYou were overwhelmed by your defeat!{PAUSE_UNTIL_PRESS}"),
+#else
+    [STRINGID_PLAYERWHITEOUT2]                      = COMPOUND_STRING("You were overwhelmed by your defeat!{PAUSE_UNTIL_PRESS}"),
+#endif
```
Comment (AlexOn1ine) [inline src/battle_message.c:194]: We don't have any other configs for messages. Does it make sense to have one here?

**Review by AlexOn1ine — COMMENTED** [review]
2 minor things but looks good otherwise

File: src/battle_message.c (hunk 1 above)
Comment (AsparagusEduardo) [reply src/battle_message.c:194]: This is specifically for the scaling money feature upon whiteout from FRLG/Gen4+. In Gen 1/2/RSE, you instead lose half your money without the game displaying how much it was that you lost.

File: src/battle_message.c [hunk 2]
```diff
@@ -470,1121 +143,747 @@ const u8 *const gStatNamesTable[NUM_BATTLE_STATS] =
...
+    [STRINGID_PKMNISPARALYZED]                      = COMPOUND_STRING("{B_ATK_NAME_WITH_PREFIX} couldn't move because it's paralyzed!"),
+    [STRINGID_PKMNISALREADYPARALYZED]               = COMPOUND_STRING("{B_DEF_NAME_WITH_PREFIX} is already paralyzed!"),
+    [STRINGID_PKMNHEALEDPARALYSIS]                  = COMPOUND_STRING("{B_DEF_NAME_WITH_PREFIX} was cured of paralysis!"),
+    [STRINGID_PKMNDREAMEATEN]                       = COMPOUND_STRING("{B_DEF_NAME_WITH_PREFIX}'s dream was eaten!"), //not in gen 5+, expansion doesn't use anymore
+    [STRINGID_STATSWONTINCREASE]                    = COMPOUND_STRING("{B_ATK_NAME_WITH_PREFIX}'s {B_BUFF1} won't go any higher!"),
+    [STRINGID_STATSWONTDECREASE]                    = COMPOUND_STRING("{B_DEF_NAME_WITH_PREFIX}'s {B_BUFF1} won't go any lower!"),
+    [STRINGID_TEAMSTOPPEDWORKING]                   = COMPOUND_STRING("Your team's {B_BUFF1} stopped working!"), //unused
+    [STRINGID_FOESTOPPEDWORKING]                    = COMPOUND_STRING("The foe's {B_BUFF1} stopped working!"), //unused
```
[the remark this answers is not in the document]
Comment (AsparagusEduardo) [reply src/battle_message.c:226]: Sure, let me do that

[the remark this answers is not in the document]
Comment (AsparagusEduardo) [reply src/battle_message.c:226]: @AlexOn1ine thoughts?
I feel this is a non-blocking issue.

[the remark this answers is not in the document]
Comment (AlexOn1ine) [reply src/battle_message.c:226]: It isn't blocking for me so I let you both decide.

[the remark this answers is not in the document]
Comment (AsparagusEduardo) [reply src/battle_message.c:226]: Alright, let's keep it for now.

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**